v3/src/components/graph/adornments/movable-line.tsx

```diff
@@ -1,100 +1,105 @@
+              y: y(pointsOnAxes.current.pt2.y)
+            }
+          },
+          breakPt1 = {
+            x: pixelPtsOnAxes.pt1.x + (3 / 8) * (pixelPtsOnAxes.pt2.x - pixelPtsOnAxes.pt1.x),
+            y: pixelPtsOnAxes.pt1.y + (3 / 8) * (pixelPtsOnAxes.pt2.y - pixelPtsOnAxes.pt1.y)
           },
-          pt2: {
-            x: x(pointsOnAxes.pt2.x),
-            y: y(pointsOnAxes.pt2.y)
-          }
-        },
-        breakPt1 = {
-          x: pixelPtsOnAxes.pt1.x + (3 / 8) * (pixelPtsOnAxes.pt2.x - pixelPtsOnAxes.pt1.x),
-          y: pixelPtsOnAxes.pt1.y + (3 / 8) * (pixelPtsOnAxes.pt2.y - pixelPtsOnAxes.pt1.y)
-        },
-        breakPt2 = {
-          x: pixelPtsOnAxes.pt1.x + (5 / 8) * (pixelPtsOnAxes.pt2.x - pixelPtsOnAxes.pt1.x),
-          y: pixelPtsOnAxes.pt1.y + (5 / 8) * (pixelPtsOnAxes.pt2.y - pixelPtsOnAxes.pt1.y)
-        },
-        endPointsArray = [
-          {pt1: pixelPtsOnAxes.pt1, pt2: pixelPtsOnAxes.pt2},
-          {pt1: pixelPtsOnAxes.pt1, pt2: breakPt1},
-          {pt1: breakPt1, pt2: breakPt2},
-          {pt1: breakPt2, pt2: pixelPtsOnAxes.pt2}
-        ]
-      fixEndPoints(lineObject.line, 0)
-      fixEndPoints(lineObject.lower, 1)
-      fixEndPoints(lineObject.middle, 2)
-      fixEndPoints(lineObject.upper, 3)
-
-      refreshEquation()
-    }, [pointsOnAxes, lineObject, props.transform, slope, intercept, x, xDomain, y, yDomain]
+          breakPt2 = {
+            x: pixelPtsOnAxes.pt1.x + (5 / 8) * (pixelPtsOnAxes.pt2.x - pixelPtsOnAxes.pt1.x),
+            y: pixelPtsOnAxes.pt1.y + (5 / 8) * (pixelPtsOnAxes.pt2.y - pixelPtsOnAxes.pt1.y)
+          },
+          endPointsArray = [
+            {pt1: pixelPtsOnAxes.pt1, pt2: pixelPtsOnAxes.pt2},
+            {pt1: pixelPtsOnAxes.pt1, pt2: breakPt1},
+            {pt1: breakPt1, pt2: breakPt2},
+            {pt1: breakPt2, pt2: pixelPtsOnAxes.pt2}
+          ]
+        fixEndPoints(lineObject.line, 0)
+        fixEndPoints(lineObject.lower, 1)
+        fixEndPoints(lineObject.middle, 2)
+        fixEndPoints(lineObject.upper, 3)
+
+        refreshEquation()
+      })
+      return () => disposer()
+    }, [pointsOnAxes, lineObject, props.transform, x, xDomain, y, yDomain, model]
   )
 
   const
     continueTranslate = useCallback((event: MouseEvent) => {
       const tWorldX = x.invert(event.x - 60),
         tWorldY = y.invert(event.y)
-      setLine({slope, intercept: tWorldY - slope * tWorldX})
-    }, [setLine, slope, x, y]),
+      model.setLine({slope: model.slope, intercept: tWorldY - model.slope * tWorldX})
+    }, [model, x, y]),
 
     continueRotation1 = useCallback((event: { x: number, y: number, dx: number, dy: number }) => {
       if (event.dx !== 0 || event.dy !== 0) {
         let isVertical = false
         const newPivot1 = {x: x.invert(event.x - 60), y: y.invert(event.y)},
-          pivot2 = props.line.pivot2 || pointsOnAxes.pt2
+          pivot2 = isFinite(model.pivot2.x) ? model.pivot2 : pointsOnAxes.current.pt2
```

Comment:
You could also add an `isValid()` view on the `PointModel` and use `model.pivot2.isValid`.